src/internal/storage/fileset/storage.go

```diff
@@ -506,3 +506,22 @@ func (s *Storage) CreateChunkSet(ctx context.Context, tx *sqlx.Tx) (ChunkSetID,
 
 	return chunksetID, nil
 }
+
+func (s *Storage) DropChunkSet(ctx context.Context, tx *sqlx.Tx, id ChunkSetID) error {
+	result, err := tx.Exec("DELETE FROM storage.chunksets WHERE id = $1", id)
+	if err != nil {
+		return errors.Wrap(err, "delete chunkset in db")
+	}
+	// Check the number of affected rows
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return errors.Wrap(err, "rows effected")
+	}
+	if rowsAffected == 0 {
+		return errors.Errorf("no chunkset found with the given id: %d", id)
+	}
+
+	// encode chunkset into string for tracker
+	strID := "chunkset/" + strconv.FormatUint(uint64(id), 10)
```

Comment:
[nit]: It would be good to write a tiny function that does this, which can help inline some of this code and make it more compact. Compact code can sometimes help improve readability.